### ISSUE TITLE
Suspicious Comments - Possible false positive?

### ISSUE THREAD

**Issue by omerlh** [zaproxy/zaproxy#6622]
**Describe the bug**
The Suspicious Comments passive scan rule detect strings, not only comments, for example:

```html
<html>
    <head>
        <script>
            s = 'user'
        </script>
    </head>
    <body>
        <h1> hello </h1>
    </body>
</html>
```

It will match on the line `s = 'user'` even though it is not a comment

**To Reproduce**
1. Serve the file bellow (for example `npx serve` + `ngrok http 5000`)
2. Scan the target with Zap
3. See the Suspicious Comments alert:

```
The following pattern was used: \bUSER\b and was detected in the element starting with: "<script>
            s = 'user'
        </script>", see evidence field for the suspicious comment/snippet.
```

**Expected behavior**
No alert is raised 

**Software versions**
![image](https://user-images.githubusercontent.com/6730584/119302395-b8ed9580-bc6c-11eb-928a-ddb222149387.png)

**Comment by thc202** [comment]
The scan rule does not check the comments in JavaScript but the whole content.

**Comment by omerlh** [comment]
Yep, I noticed that from the code - is this intentional?

**Comment by thc202** [comment]
Yes, lack of JavaScript parser to extract/analyse just the comments.

**Comment by kingthorin** [comment]
I don’t think there’s a way around this, unless someone knows of a library that also handles minified JS.
